app/modules/net_info.c

```diff
@@ -0,0 +1,176 @@
+static int ping_host_count;
+static ip_addr_t ping_host_ip;
+
+
+void ping_received(void *arg, void *data) {
+    // this would require change of the interface 
+    // struct ping_msg *pingmsg = (struct ping_msg*)arg;
+    // struct ping_option *pingopt = pingmsg->ping_opt;
+    struct ping_option *pingopt = (struct ping_option*)arg;
+    struct ping_msg *pingmsg = pingopt->parent_msg ; 
+
+    struct ping_resp *pingresp = (struct ping_resp*)data;
+
+    char ipaddrstr[16];
+    ip_addr_t source_ip;
+    
+    source_ip.addr = pingopt->ip;
+    ipaddr_ntoa_r(&source_ip, ipaddrstr, sizeof(ipaddrstr));
+
+    if (ping_callback_ref != LUA_NOREF) {
+      lua_rawgeti(gL, LUA_REGISTRYINDEX, ping_callback_ref);
+      lua_pushinteger(gL, pingresp->bytes);
+      lua_pushstring(gL, ipaddrstr);
+      lua_pushinteger(gL, pingresp->seqno);
+      lua_pushinteger(gL, pingresp->ttl);
+      lua_pushinteger(gL, pingresp->resp_time);
+      lua_call(gL, 5, 0);
+    } else {
+      c_printf("%d bytes from %s, icmp_seq=%d ttl=%d time=%dms\n",
+	       pingresp->bytes,
+	       ipaddrstr,
+	       pingresp->seqno,
+	       pingresp->ttl,
+	       pingresp->resp_time);
+    }
+}
+
+
+
+static void ping_by_hostname(const char *name, ip_addr_t *ipaddr, void *arg) {
+    struct ping_option *ping_opt = (struct ping_option *)c_zalloc(sizeof(struct ping_option));
+
+    if (ipaddr == NULL) {
+      c_printf("SEVERE problem resolving hostname - network and DNS accessible?\n");
+        return;
+    }
+    if (ipaddr->addr == IPADDR_NONE) {
+      c_printf("problem resolving hostname - maybe nonexistent host?\n");
+	return;
+    }
+    
+    ping_opt->count = ping_host_count;
+    ping_opt->ip = ipaddr->addr;
+    ping_opt->coarse_time = 0;
+    ping_opt->recv_function = &ping_received;
+    
+    ping_start(ping_opt);
+}
+
+
+/**
+  * test.ping()
+  * Description:
+  * 	Send ICMP ping request to address, optionally call callback when response received
+  *
+  * Syntax:
+  * 	wifi.sta.getconfig(ssid, password) --Set STATION configuration, Auto-connect by default, Connects to any BSSID
+  *     test.ping(address)              -- send 4 ping requests to target address
+  *     test.ping(address, n)           -- send n ping requests to target address
+  *     test.ping(address, n, callback) -- send n ping requests to target address
+  * Parameters:
+  * 	address: string 
+  * 	n: number of requests to send
+  * 	callback: 
+  * Returns:
+  * 	Nothing.
+  *
+  * Example:
+  *     test.ping("192.168.0.1")               -- send 4 pings to 192.168.0.1
+  *     test.ping("192.168.0.1", 10)           -- send 10 pings to 192.168.0.1
+  *     test.ping("192.168.0.1", 10, got_ping) -- send 10 pings to 192.168.0.1, call got_ping() with the
+  *                                           --     ping results
+  */
+
+static int net_info_ping(lua_State *L)
+{
+    const char *ping_target;
+    unsigned count = 4;
+
+    // retrieve address arg (mandatory)
+    if (lua_isstring(L, 1)) {
```

Comment:
Since you're not checking for any other type of argument 1:
The whole if statement can be replaced by
```c
ping_target = luaL_checkstring(L, 1);
```